src/nccl_ofi_sendrecv.c

```diff
@@ -1595,7 +1595,11 @@ static int listen(nccl_net_ofi_ep_t *base_ep,
 	/* Build handle */
 	local_ep_name = get_local_address(ep->ofi_ep);
 
-	memcpy(handle->ep_name, local_ep_name, MAX_EP_ADDR);
+	if (local_ep_name != NULL) {
+		memcpy(handle->ep_name, local_ep_name, MAX_EP_ADDR);
+		free(local_ep_name);
```

Comment:
Blegh, late night, thanks. Still, this is not quite right. We should either use the name in the handle or hold on to a reference somewhere, no?

Reply:
Yes, there's definitely a leak. Ideally, we would just have `get_local_address` write to a pointer instead of allocating memory. You can also fix this code by using `handle->ep_name` instead of `local_ep_name` after it has already been freed.